activerecord/lib/active_record/associations/collection_proxy.rb

```diff
@@ -728,8 +728,8 @@ def distinct
       #   #       #<Pet id: 2, name: "Spook", person_id: 1>,
       #   #       #<Pet id: 3, name: "Choo-Choo", person_id: 1>
       #   #    ]
```

Comment:
This doc also says we're using SQL counting. We should also have an example with a block.